integration/integration_test.go

```diff
@@ -313,3 +313,66 @@ func TestDemoteLeader(t *testing.T) {
 	numManager--
 	pollClusterReady(t, cl, numWorker, numManager)
 }
+
+func TestJoinDemoteRace(t *testing.T) {
```

Comment:
The iterations were for me to try to trigger this particular condition, but I could take it out and it should trigger it, just less often.  I will also try to do a forced removal.